torchao/quantization/observer.py

```diff
@@ -130,8 +160,13 @@ def forward(self, input: torch.Tensor):
             return input
 
         input_detached = input.detach()
-        if self.block_size is None:
-            self.block_size = get_block_size(input_detached.shape, self.granularity_type)
+        assert self.granularity_type is not None, "granularity_type is None"
+        block_size = get_block_size(input_detached.shape, self.granularity_type)
+
+        # If we are doing PerTensor quantization then we do not need to reduce along a given
+        # Dimension. The input tensor can vary sizes and we dont want to cache a block_size
+        if self.block_size is None and not isinstance(self.granularity_type, PerTensor):
```

Comment:
I think it probably makes sense to remove `self.block_size` and always recalculate the block_size

Reply:
you are right I thought it was needed in choose_qparams, but we are good